lib/connector/OAuthConnector.js

```diff
@@ -150,6 +150,42 @@ class OAuthConnector {
         throw new Error('Please implement the getUserProfile and getUserId methods in the class deriving from OAuthConnector.');
     }
 
+    /**
+     * Called after successful completion of the connector's configuration flow. The 'data' parameter contains
+     * configuration properties generated by settings managers that ran prior to this connector, for example a user ID
+     * of the user in another system, or a URL to obtain the access token to another system. You can use this extensibility
+     * point to modify the 'userContext' with information about the identity of the user in another system, therefore
+     * creating an association between the same user in two systems.
+     * @param {FusebitContext} fusebitContext The Fusebit context of the request
+     * @param {*} userContext The user context representing the vendor's user. Contains vendorToken and vendorUserProfile, representing responses
+     * from getAccessToken and getUserProfile, respectively.
+     * @param {*} data A property bag containing properties generated by settings managers that have completed prior to the configuration flow of this connector.
+     */
+    async onConfigurationComplete(fusebitContext, userContext, data) {
+        if (data) {
+            for (let p in data) {
+                const match = p.match(/^(.+)_oauth_user_id$/);
+                if (match && typeof data[`${match[1]}_oauth_connector_base_url`] === 'string') {
+                    userContext.foreignOAuthIdentities = userContext.foreignOAuthIdentities || {};
```

Comment:
```
userContext.foreignOAuthIdentities = {
    ...(userContext.foreignOAuthIdentities || {}),
    [match[1]]: { ... }
};
```

Reply:
I was staring at it for a while, there is no bug, is there? I think the code in there accomplishes the same result in the same number of lines?

Reply:
No bug, however a single assignment composition is clearer than several assignment operations.

Reply:
Done